flytekit/experimental/eager_function.py

```diff
@@ -533,8 +542,10 @@ async def wrapper(*args, **kws):
                 await cleanup_fn()
 
     secret_requests = kwargs.pop("secret_requests", None) or []
-    if client_secret_group is not None and client_secret_key is not None:
+    try:
         secret_requests.append(Secret(group=client_secret_group, key=client_secret_key))
+    except ValueError:
+        pass
```

Comment:
what if there's nothing in the client_secret_env_var environment variable?

Reply:
`client_secret_env_var` doesn't figure into this try except block, or am I missing something?

Reply:
no yeah i know, i mean, what happens if all three (`client_secret_env_var`, `client_secret_group` and `client_secret_key`) are None?